lib/cdo/contact_rollups.rb

```diff
@@ -808,11 +808,10 @@ def self.update_data_from_forms(form_kind)
         raise
       end
 
-      # Discard the connection and get a new one after each batch. Currently, if you use multiple statements in a batch, the first call will succeed; the next call on
-      # the same connection AFTER you do a multi-statement call will fail with a MySQL error that commands are out of order. The innertubes suggest that this may be a problem in mysql2 gem.
-      # Fortunately, it is fairly cheap to get a new connection for each batch (time to get new connection is negligible compared to batch time).
-      conn.disconnect
-      conn = mysql_multi_connection
+      # https://www.rubydoc.info/github/brianmario/mysql2/Mysql2%2FClient:abandon_results!
+      # When using MULTI_STATEMENTS support, calling this will throw away any unprocessed results as fast as it can in
+      # order to put the connection back into a state where queries can be issued again.
+      conn.abandon_results!
```

Comment:
Reading more closely, `abandon_results!` is a method on the MySQL `Client` Class.  In this part of ContactRollups, `conn` is an instance of `Sequel::Mysql2::Database`.

Reply:
Looks like the syntax would be `conn.pool.hold(&:abandon_results!)` to invoke this command directly on the MySQL client that Sequel is using -